scipy/optimize/_highs/meson.build

```diff
@@ -1,290 +1,75 @@
-    '../../_lib/highs/src/util/HFactorRefactor.cpp',
-    '../../_lib/highs/src/util/HFactorUtils.cpp',
-    '../../_lib/highs/src/util/HighsHash.cpp',
-    '../../_lib/highs/src/util/HighsLinearSumBounds.cpp',
-    '../../_lib/highs/src/util/HighsMatrixPic.cpp',
-    '../../_lib/highs/src/util/HighsMatrixUtils.cpp',
-    '../../_lib/highs/src/util/HighsSort.cpp',
-    '../../_lib/highs/src/util/HighsSparseMatrix.cpp',
-    '../../_lib/highs/src/util/HighsUtils.cpp',
-    '../../_lib/highs/src/util/HSet.cpp',
-    '../../_lib/highs/src/util/HVectorBase.cpp',
-    '../../_lib/highs/src/util/stringutil.cpp',
-    '../../_lib/highs/src/interfaces/highs_c_api.cpp'
-  ],
-  include_directories: [
-    'src/',
-    '../../_lib/highs/extern/',
-    '../../_lib/highs/src/',
-    '../../_lib/highs/src/io/',
-    '../../_lib/highs/src/ipm/ipx/include/',
-    '../../_lib/highs/src/lp_data/',
-    '../../_lib/highs/src/util/',
-  ],
-  dependencies: thread_dep,
-  cpp_args: [highs_flags, highs_define_macros],
-  gnu_symbol_visibility: 'inlineshidden',
+_highslib = static_library(
+    'highs',
+    highslib_srcs,
+    dependencies: highs_deps,
+    cpp_args: highs_args,
+    c_args: highs_args,
+    include_directories: highs_incdirs,
+    # Ensure that if we link a static library into a shared library,
+    # private symbols don't get re-exported.
+    gnu_symbol_visibility: 'inlineshidden',
+    pic: true,
+    install: true,
+    install_dir: py3.get_install_dir() / 'scipy/optimize/_highspy',
 )
 
-_highs_wrapper = py3.extension_module('_highs_wrapper',
-  cython_gen_cpp.process('cython/src/_highs_wrapper.pyx'),
-  include_directories: [
-    'cython/src/',
-    'src/',
-    '../../_lib/highs/src/',
-    '../../_lib/highs/src/io/',
-    '../../_lib/highs/src/lp_data/',
-    '../../_lib/highs/src/util/'
-  ],
-  dependencies: [np_dep, thread_dep, atomic_dep],
-  link_args: version_link_args,
-  link_with: [highs_lib, ipx_lib, basiclu_lib],
-  cpp_args: [highs_flags, highs_define_macros, cython_c_args],
-  install: true,
-  subdir: 'scipy/optimize/_highs'
+_highs_dep = declare_dependency(
+    link_with: _highslib,
+    dependencies: highs_deps,
+    include_directories: highs_incdirs,
 )
 
-_highs_constants = py3.extension_module('_highs_constants',
-  cython_gen_cpp.process('cython/src/_highs_constants.pyx'),
-  c_args: cython_c_args,
-  include_directories: [
-    'cython/src/',
-    'src',
-    '../../_lib/highs/src/',
-    '../../_lib/highs/src/io/',
-    '../../_lib/highs/src/lp_data/',
-    '../../_lib/highs/src/simplex/'
-  ],
-  dependencies: thread_dep,
-  link_args: version_link_args,
-  install: true,
-  subdir: 'scipy/optimize/_highs'
+highspy_cpp = highs_proj.get_variable('highspy_cpp')
+highs_conf_file = highs_proj.get_variable('highs_conf_file')
+
+scipy_highspy_dep = [py3_dep, pybind11_dep, _highs_dep, thread_dep, atomic_dep]
+
+py3.install_sources(
+    ['__init__.py', '_highs_wrapper.py'],
+    subdir: 'scipy/optimize/_highspy',
```

Comment:
Why have the directory be `optimize/_highs/` and install everything into `optimize/_highspy/`? That seems unnecessarily confusing, I'd choose one or the other.

Reply:
Decided to go with `highspy` throughout.